### PR TITLE
Feature/rpc improvment 02

### PR DESCRIPTION
**Pull request by xrdavies**
* add rpc method xdag_get_block_info

based on #304

### CODE REVIEW

File: client/block.c [hunk 1]
```diff
@@ -1445,9 +1445,8 @@ static int bi_compar(const void *l, const void *r)
 }
 
 // returns string representation for the block state. Ignores BI_OURS flag
-static const char* get_block_state_info(struct block_internal *block)
+const char* xdag_get_block_state_info(uint8_t flag)
 {
-	const uint8_t flag = block->flags & ~BI_OURS;
```
Comment (jonano614) [inline client/block.c]: I would recommend you to leave this line. 
`& ~BI_OURS` on each call of function does not looke good.

File: client/block.c [hunk 2]
```diff
@@ -1783,13 +1782,11 @@ int xdag_get_transactions(xdag_hash_t hash, void *data, int (*callback)(void*, i
 	for (i = 0; i < n; ++i) {
 		if (!i || block_array[i] != block_array[i - 1]) {
 			struct block_internal *ri = block_array[i];
-			if (ri->flags & BI_APPLIED) {
```
Comment (jonano614) [inline client/block.c]: Is this condition not necessary?

Comment (xrdavies) [reply client/block.c]: Not necessary, to query transactions users need to know the status of the tx block even the block is pending or rejected.

**Comment by jonano614** [comment]
Could youl please resolve conflicts?

**Comment by xrdavies** [comment]
@jonano614  Conflicts resolved.

**Pull request merged**